packages/hub-threads-client/src/client.ts

```diff
@@ -35,11 +35,16 @@ Client.prototype.getThread = async function (name: string, ctx?: Context): Promi
       .then((meta) => {
         client.getThread(req, meta, (err: ServiceError | null, message: GetThreadResponse | null) => {
           if (err) reject(err)
-          const msg = message?.toObject()
-          if (msg) {
-            msg.id = ThreadID.fromBytes(Buffer.from(msg.id as string, 'base64')).toString()
+          if (message) {
+            const res = {
+              name: message.getName(),
+              isDb: message.getIsDb(),
+              id: ThreadID.fromBytes(message.getId_asU8()).toString(),
+            }
+            resolve(res)
+          } else {
+            resolve()
```

Comment:
This is fine (correct if you ask me), but if/when we update Typescript, its not going to like this :(

Reply:
so, it was `resolve(undefined)` before this change... what's the solution do you think?

Reply:
I think what you have is good. I had to fight typescript over in js-threads, because the promise return type doesn't include void or undefined technically. So I guess we should throw in that case? But some of the responses _do_ return void... so long story short... I'm not sure.

Reply:
want to sneak in a no result error?

Reply:
added